Timestamp not handled correctly

I have following issue with timestamp and mypy:
#### `mwe.py`
```python
import proto
from google.protobuf import timestamp_pb2


class MyClass(proto.Message):
    create_time = proto.Field(
        proto.MESSAGE,
        number=6,
        message=timestamp_pb2.Timestamp,
    )


if __name__ == '__main__':
    classes = [MyClass(create_time=timestamp_pb2.Timestamp(seconds=1, nanos=1)),
               MyClass(create_time=timestamp_pb2.Timestamp(seconds=2, nanos=2))]

    print(f"{max(classes, key=lambda m: m.create_time)=}")
```
when I run the code, it succeeds:
```shell
python .\mwe.py            
max(classes, key=lambda m: m.create_time)=create_time {
  seconds: 2
  nanos: 2
}
```
but it fails on mypy
```shell
mypy .\mwe.py
mwe.py:17: error: Argument "key" to "max" has incompatible type "Callable[[MyClass], Timestamp]"; expected "Callable[[MyClass], Union[SupportsDunderLT[Any], SupportsDunderGT[Any]]]"  [arg-type]
mwe.py:17: error: Incompatible return value type (got "Timestamp", expected "Union[SupportsDunderLT[Any], SupportsDunderGT[Any]]")  [return-value]
Found 2 errors in 1 file (checked 1 source file)
```

I hope this is the right package, because this is an issue between [mypy](https://github.com/python/mypy), [protobuf](https://github.com/protocolbuffers/protobuf/blob/main/python/README.md), [proto-plus](https://github.com/googleapis/proto-plus-python) and maybe something else.
When I have a `Timestamp` as a field, the `proto-plus` turns it into [`DatetimeWithNanoseconds`](https://github.com/googleapis/proto-plus-python/blob/v1.23.0/proto/datetime_helpers.py#L76) which has arithmetic defined, because it extends `datetime.datetime`, but mypy does not know about it. 
This is more elaborate example
#### `problem_example.py`
```python
"""
conda create -n mypy_experiments python=3.9
conda activate mypy_experiments
"""
from datetime import datetime

import proto
from google.protobuf import timestamp_pb2


class MyClass(proto.Message):
    create_time = proto.Field(
        proto.MESSAGE,
        number=6,
        message=timestamp_pb2.Timestamp,
    )


if __name__ == '__main__':
    classes = [MyClass(create_time=timestamp_pb2.Timestamp(seconds=1, nanos=1)),
               MyClass(create_time=timestamp_pb2.Timestamp(seconds=2, nanos=2))]

    print(f"{max(classes, key=lambda m: m.create_time)=}")

    times = [d.create_time for d in classes]
    print(f"{max(times)=}")
    print(f"{type(times[0])=}, {isinstance(times[0], datetime)=}")
    times2 = [timestamp_pb2.Timestamp(seconds=1, nanos=1), timestamp_pb2.Timestamp(seconds=2, nanos=2)]
    print(f"{type(times2[0])=}, {isinstance(times2[0], datetime)=}")
    try:
        print(f"{max(times2)=}")
    except Exception as e:
        print(e)
    print(f"{max(times, key=lambda t: t)=}")
    try:
        print(f"{max(times2, key=lambda t: t)=}")
    except Exception as e:
        print(e)
```
which prints
```shell
python .\problem_example.py
max(classes, key=lambda m: m.create_time)=create_time {
  seconds: 2
  nanos: 2
}

max(times)=DatetimeWithNanoseconds(1970, 1, 1, 0, 0, 2, tzinfo=datetime.timezone.utc)
type(times[0])=<class 'proto.datetime_helpers.DatetimeWithNanoseconds'>, isinstance(times[0], datetime)=True
type(times2[0])=<class 'google.protobuf.timestamp_pb2.Timestamp'>, isinstance(times2[0], datetime)=False
'>' not supported between instances of 'Timestamp' and 'Timestamp'
max(times, key=lambda t: t)=DatetimeWithNanoseconds(1970, 1, 1, 0, 0, 2, tzinfo=datetime.timezone.utc)
'>' not supported between instances of 'Timestamp' and 'Timestamp'
```

I am using python 3.9 and here are all versions of packages:
```shell
pip freeze
filelock==3.13.1
mypy==1.8.0
mypy-extensions==1.0.0
mypy-protobuf==3.5.0
proto-plus==1.23.0
proto-plus-stubs==0.6.0        
protobuf==4.25.2
requests-file==1.5.1
tldextract==5.1.1
tomli==2.0.1
types-protobuf==4.24.0.20240106
typing_extensions==4.9.0       
XlsxWriter==3.1.9
```

--- a/proto-stubs/fields.pyi
+++ b/proto-stubs/fields.pyi
@@ -9,6 +9,9 @@ from typing import (
     overload,
 )
 
+from google.protobuf.timestamp_pb2 import Timestamp
+
+from .datetime_helpers import DatetimeWithNanoseconds
 from .message import Message
 from .primitives import ProtoType
 
@@ -90,6 +93,17 @@ class Field(Generic[_T]):
     ) -> None: ...
     @overload
     def __init__(
+        self: Field[DatetimeWithNanoseconds],
+        proto_type: Literal[ProtoType.MESSAGE],
+        *,
+        number: int,
+        message: type[Timestamp],
+        oneof: str | None = None,
+        json_name: str | None = None,
+        optional: bool = False,
+    ) -> None: ...
+    @overload
+    def __init__(
         self: Field[_T],
         proto_type: Literal[ProtoType.MESSAGE],
         *,
@@ -106,6 +120,16 @@ class Field(Generic[_T]):
         *,
         number: int,
         enum: type[_T],
+        oneof: str | None = None,
+        json_name: str | None = None,
+        optional: bool = False,
+    ) -> None: ...
+    @overload
+    def __init__(
+        self: Field[DatetimeWithNanoseconds],
+        proto_type: type[Timestamp],
+        *,
+        number: int,
         oneof: str | None = None,
         json_name: str | None = None,
         optional: bool = False,
@@ -206,6 +230,17 @@ class RepeatedField(Field[_T]):
     ) -> None: ...
     @overload
     def __init__(
+        self: RepeatedField[DatetimeWithNanoseconds],
+        proto_type: Literal[ProtoType.MESSAGE],
+        *,
+        number: int,
+        message: type[Timestamp],
+        oneof: str | None = None,
+        json_name: str | None = None,
+        optional: bool = False,
+    ) -> None: ...
+    @overload
+    def __init__(
         self: RepeatedField[_T],
         proto_type: Literal[ProtoType.MESSAGE],
         *,
@@ -222,6 +257,16 @@ class RepeatedField(Field[_T]):
         *,
         number: int,
         enum: type[_T],
+        oneof: str | None = None,
+        json_name: str | None = None,
+        optional: bool = False,
+    ) -> None: ...
+    @overload
+    def __init__(
+        self: RepeatedField[DatetimeWithNanoseconds],
+        proto_type: type[Timestamp],
+        *,
+        number: int,
         oneof: str | None = None,
         json_name: str | None = None,
         optional: bool = False,
@@ -307,6 +352,15 @@ class MapField(Field[_V], Generic[_K, _V]):
     ) -> None: ...
     @overload
     def __init__(
+        self: MapField[int, DatetimeWithNanoseconds],
+        key_type: _IntegerProtoType,
+        value_type: Literal[ProtoType.MESSAGE],
+        *,
+        number: int,
+        message: type[Timestamp],
+    ) -> None: ...
+    @overload
+    def __init__(
         self: MapField[int, _V],
         key_type: _IntegerProtoType,
         value_type: Literal[ProtoType.MESSAGE],
@@ -322,6 +376,14 @@ class MapField(Field[_V], Generic[_K, _V]):
         *,
         number: int,
         enum: type[_V],
+    ) -> None: ...
+    @overload
+    def __init__(
+        self: MapField[int, DatetimeWithNanoseconds],
+        key_type: _IntegerProtoType,
+        value_type: type[Timestamp],
+        *,
+        number: int,
     ) -> None: ...
     @overload
     def __init__(
@@ -391,6 +453,15 @@ class MapField(Field[_V], Generic[_K, _V]):
     ) -> None: ...
     @overload
     def __init__(
+        self: MapField[str, DatetimeWithNanoseconds],
+        key_type: Literal[ProtoType.STRING],
+        value_type: Literal[ProtoType.MESSAGE],
+        *,
+        number: int,
+        message: type[Timestamp],
+    ) -> None: ...
+    @overload
+    def __init__(
         self: MapField[str, _V],
         key_type: Literal[ProtoType.STRING],
         value_type: Literal[ProtoType.MESSAGE],
@@ -406,6 +477,14 @@ class MapField(Field[_V], Generic[_K, _V]):
         *,
         number: int,
         enum: type[_V],
+    ) -> None: ...
+    @overload
+    def __init__(
+        self: MapField[str, DatetimeWithNanoseconds],
+        key_type: Literal[ProtoType.STRING],
+        value_type: type[Timestamp],
+        *,
+        number: int,
     ) -> None: ...
     @overload
     def __init__(

